### ISSUE TITLE
MacOS support

### ISSUE THREAD

**Issue by ingig** [PLangHQ/plang#4]
Confirm and fix error that come up on MacOS

**Comment by ingig** [comment]
Thanks for letting me know. To bad it went the way it went. I unfortunately haven´t been able to try it on MacOS, I was trusting that it works same on MacOS as it does on Linux, but I need to do some proper testing. Sad also about not getting into Discord, but you are welcome email me any time plang@plang.is if you take a look into it again when I have solve the MacOS to run

**Comment by RossAddinall** [comment]
@ingig if you do get a chance to do something for MacOS I'd be happy to give it a go, though I'm busy with work commitments so might not respond quickly.  There were a couple of issues that I ran into and forgive me some of these are pretty lame and yes I could go do more digging, but don't have time and feel it better the give you what I do have:
1. I used ```sudo unzip plang-osx-x64.zip -d /usr/local/plang``` to unpack it. There are two issues I faced with this, one was that ```plang``` didn't have execute permissions so I had to ```sudo chmod 755``` and the other was that this approach meant that the executable was ```/usr/local/plang/plang/plang```, I should have done ```sudo unzip plang-osx-x64.zip -d /usr/local``` and remove the redundant directory layer.
2. I managed to get around launching plang by following [this](https://support.apple.com/en-gb/guide/mac-help/mh40616/mac), but then I ran into the same issue for ```libhostfxr.dylib```  (“libhostfxr.dylib” can’t be opened because Apple cannot check it for malicious software.), at this point it felt like I was going to have to go through every exe and lib file in the distribution.

I hope this helps a little.

**Comment by ingig** [comment]
Thank you for all the info, it will for sure help. I got a friend today to run it on MacOS and it is nightmare to get it running. I'm not sure what they did so I can't give instructions, but it's so bad that I think it wouldn't be worth to post instructions

I will put in effort to fix this, although I expect it will take some time (weeks), but I'll post here when it works

The long time estimation is because I'm lost here and need to learn, so big unknown.

**Comment by RossAddinall** [comment]
Yeah,  I've just switched to MacOS (from Windows).  In some regards it's great because at the command line it's like Unix, but then you get hit with stuff like this and I have no idea what to do At least on Windows you are under no illusions and if you need Unix can spin up a VM or a container.

I think you need to somehow get your code "blessed" by Apple, which probably means paying to be an Apple developer and then submitting your code to them, but I don't really know.

**Comment by ingig** [comment]
@RossAddinall I believe I have solve the MacOS issue with running it. I managed to run it on a virtual server as I don't have a Mac my self. You can find the new release here https://github.com/PLangHQ/plang/releases (0.1.0.3)  and installation instructions https://github.com/PLangHQ/plang/blob/main/Documentation/Install.md#installing-plang-on-macos
